so_pysm_models/alms.py

```diff
@@ -76,13 +115,32 @@ def signal(self, nu=[148.], output_units="uK_RJ", **kwargs):
             nnu = 1
             nu = np.array([nu])
 
+        try:
+            output_map = self.output_map
+        except AttributeError:
+            if fwhm_arcmin is None:
+                alm = self.alm
+            else:
+                alm = hp.smoothalm(
+                    self.alm, fwhm=np.radians(fwhm_arcmin / 60), pol=True, inplace=False
+                )
+
+            output_map = self.compute_output_map(alm)
+
         # use tile to output the same map for all frequencies
-        out = np.tile(self.output_map, (nnu, 1, 1))
+        out = np.tile(output_map, (nnu, 1, 1))
         if self.wcs is not None:
             out = enmap.enmap(out, self.wcs)
-        out = out * pysm.convert_units(self.input_units, output_units, nu).reshape(
-            (nnu, 1, 1)
-        ).astype(float)
+        out *= (
+            (
+                pysm.convert_units(
+                    self.input_units, "uK_CMB", self.input_reference_frequency_GHz
+                )
+                * pysm.convert_units("uK_CMB", output_units, nu)
+            )
```

Comment:
@msyriac I think there was a bug in unit conversion for inputs in uK_RJ, can you please crosscheck if this is correct?

Reply:
Could you please fix this so that the input units are assumed to be uK_CMB? @ajvanengelen can confirm this is what the units of the lensed alms will be.